RadioController.py

```diff
@@ -38,92 +37,112 @@ def isPacketLengthConst(subpacket_id):
+        return converted
+
+    def event(self, byte_list, length):  # TODO
+        converted = {0.0}  # STUB
+        return converted
+
+    def config(self, byte_list, length):  # TODO
+        converted = {0.0}  # STUB
+        return converted
+
+    def single_sensor(self, byte_list, length):  # TODO
+        converted = {0.0}  # STUB
+        return converted
+
+    def gps(self, byte_list, length):  # TODO
+        converted = {0.0}  # STUB
+        return converted
+
+    # def acknowledgement(self, byte_list, length):  # TODO ?
+    #     converted = 0.0  # STUB
+    #     return converted
+
+    def bulk_sensor(self, byte_list: List, length: int):
+        data: Dict[int, any] = {}
+
+        # TODO REVIEW/CHANGE in type refactoring: how this is required to convert from List[bytes] to List[int]
+        int_list: List[int] = [int(x[0]) for x in byte_list]
+
+        data[SubpacketEnum.TIME.value] = self.fourtoint(int_list[0:4])
+        data[SubpacketEnum.CALCULATED_ALTITUDE.value] = self.fourtofloat(int_list[4:8])  # TODO Double check it is calculated barometer altitude with firmware
+        data[SubpacketEnum.ACCELERATION_X.value] = self.fourtofloat(int_list[8:12])
+        data[SubpacketEnum.ACCELERATION_Y.value] = self.fourtofloat(int_list[12:16])
+        data[SubpacketEnum.ACCELERATION_Z.value] = self.fourtofloat(int_list[16:20])
+        data[SubpacketEnum.ORIENTATION_1.value] = self.fourtofloat(int_list[20:24])
+        data[SubpacketEnum.ORIENTATION_2.value] = self.fourtofloat(int_list[24:28])
+        data[SubpacketEnum.ORIENTATION_3.value] = self.fourtofloat(int_list[28:32])
+        data[SubpacketEnum.LONGITUDE.value] = self.fourtofloat(int_list[36:40])
+        data[SubpacketEnum.LATITUDE.value] = self.fourtofloat(int_list[32:36])
+        data[SubpacketEnum.STATE.value] = int_list[40]
+        return data
+
+
+    # Dictionary of subpacket id - function to parse that data
+    packetTypeToParser: Dict[int, Callable[[list, int], Any]] = {  # TODO review this type hint
+        SubpacketEnum.STATUS_PING.value: status_ping,
+        SubpacketEnum.MESSAGE.value: message,
+        SubpacketEnum.EVENT.value: event,
+        SubpacketEnum.CONFIG.value: config,
+        # SubpacketEnum.SINGLE_SENSOR.value: single_sensor, # Todo make this a range of keys?
+        SubpacketEnum.GPS.value: gps,
+        # SubpacketEnum.ACKNOWLEDGEMENT.value: acknowledgement, # TODO Uncomment when complete
+        SubpacketEnum.BULK_SENSOR.value: bulk_sensor,
+    }
+
+    def fourtofloat(self, bytes):
+        assert len(bytes) == 4
+        data = bytes
+        b = struct.pack('4B', *data)
+        c = struct.unpack('>f' if self.bigEndianFloats else '<f', b)
+        return c[0]
+
+    def fourtoint(self, bytes):
+        assert len(bytes) == 4
+        data = bytes
+        # data = data[::-1]#flips bytes
+        b = struct.pack('4B', *data)
+        # big endian
+        c = struct.unpack('>I' if self.bigEndianInts else '<I', b)
```

Comment:
Makes more sense!

Reply:
Oops, I should probably get rid of those old comments tho.